container_api/src/config.rs

```diff
@@ -0,0 +1,217 @@
+use holochain_core_types::{
+    entry::agent::{Agent, Identity},
+    error::{HcResult, HolochainError},
+    json::JsonString,
+};
+use holochain_dna::Dna;
+use serde::Deserialize;
+use std::{convert::TryFrom, fs::File, io::prelude::*};
+
+#[derive(Deserialize)]
+pub struct Configuration {
+    agents: Option<Vec<AgentConfiguration>>,
+    dnas: Option<Vec<DNAConfiguration>>,
+    instances: Option<Vec<InstanceConfiguration>>,
+    interfaces: Option<Vec<InterfaceConfiguration>>,
+    bridges: Option<Vec<Bridge>>,
+}
+
+#[derive(Deserialize, Clone)]
+pub struct AgentConfiguration {
+    id: String,
+    key_file: Option<String>,
+}
+
+impl Into<Agent> for AgentConfiguration {
```

Comment:
`Into<T>` is implicitly generated when you implement `From<T>` for a type. It shouldn't be necessary to implement `Into<T>`

Reply:
Done